src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingService.kt

```diff
@@ -34,4 +36,23 @@ class BookingService(
       returnToCustodyDate = prisonApiSourceData.returnToCustodyDate?.returnToCustodyDate
     )
   }
+
+  fun getCalculationBreakdown(
+    calculationRequestId: Long
+  ): CalculationBreakdown {
+    val calculationUserInputs = calculationTransactionalService.findUserInput(calculationRequestId)
+    val prisonerDetails = calculationTransactionalService.findPrisonerDetailsFromCalculation(calculationRequestId)
+    val sentenceAndOffences = calculationTransactionalService.findSentenceAndOffencesFromCalculation(calculationRequestId)
+    val bookingAndSentenceAdjustments = calculationTransactionalService.findBookingAndSentenceAdjustmentsFromCalculation(calculationRequestId)
+    val returnToCustodyDate = calculationTransactionalService.findReturnToCustodyDateFromCalculation(calculationRequestId)
+    val calculation = calculationTransactionalService.findCalculationResults(calculationRequestId)
+    val booking = Booking(
```

Comment:
Would it be possible to refactor this so the creation of the Booking is not duplicated? Maybe the validation shouldn't be within the getBooking funciton.

Reply:
Potentially.. I was going to look at that next. Didn't want to do too big a PR in one go.  Will continue the discussion.